src/Files.App/Filesystem/FilesystemOperations/FilesystemOperations.cs

```diff
@@ -880,14 +912,22 @@ public async Task<IStorageHistory> CopyItemsAsync(IList<IStorageItemWithPath> so
 			return null;
 		}
 
-		public async Task<IStorageHistory> MoveItemsAsync(IList<IStorageItem> source, IList<string> destination, IList<FileNameConflictResolveOptionType> collisions, IProgress<float> progress, IProgress<FileSystemStatusCode> errorCode, CancellationToken cancellationToken)
+		public async Task<IStorageHistory> MoveItemsAsync(IList<IStorageItem> source, IList<string> destination, IList<FileNameConflictResolveOptionType> collisions, IProgress<FileSystemProgress> progress, CancellationToken cancellationToken)
 		{
-			return await MoveItemsAsync(await source.Select((item) => item.FromStorageItem()).ToListAsync(), destination, collisions, progress, errorCode, cancellationToken);
+			return await MoveItemsAsync(await source.Select((item) => item.FromStorageItem()).ToListAsync(), destination, collisions, progress, cancellationToken);
 		}
 
-		public async Task<IStorageHistory> MoveItemsAsync(IList<IStorageItemWithPath> source, IList<string> destination, IList<FileNameConflictResolveOptionType> collisions, IProgress<float> progress, IProgress<FileSystemStatusCode> errorCode, CancellationToken token)
-		{
-			var rawStorageHistory = new List<IStorageHistory>();
+		public async Task<IStorageHistory> MoveItemsAsync(IList<IStorageItemWithPath> source, IList<string> destination, IList<FileNameConflictResolveOptionType> collisions, IProgress<FileSystemProgress> progress, CancellationToken token)
+        {
+            FileSystemProgress fsProgress = new(progress)
+            {
+                ItemsCount = source.Count,
+                EnumerationCompleted = true,
+                Status = FileSystemStatusCode.InProgress
+            };
+            fsProgress.Report();
```

Comment:
This bit of code is used a lot in through this file, could it be worth it to put it inside a builder method initializing the whole process and returning it?

Reply:
I decided to put them in the constructor.